app/src/main/java/fnsb/macstransit/Activities/SplashActivity.java

```diff
@@ -238,48 +251,92 @@ private Thread loadData() {
-
-				// Update the progress to halfway, and inform the user that we are now loading stops.
-				Log.d("loadData", "Loading stops in the routes");
-				this.setProgress(0.5d);
-				this.setMessage(R.string.load_stops);
-
-				// Load the stops in each route.
-				for (int i = 0; i < this.routes.length; i++) {
-					// Get the route at the current index (i) from all the routes that were loaded.
-					Route route = this.routes[i];
-
-					// Load the stops in the route.
-					route.stops = route.loadStops(this.routeMatch);
-
-					// Set the progress to the current index (plus 1) out of the all the routes to be parsed,
-					// and then divide that value in half, as this is the other 50% to be processed.
-					this.setProgress(0.5d + (((i + 1d) / this.routes.length) / 2));
+			// Get the master schedule from the RouteMatch server
+			org.json.JSONObject masterSchedule = this.routeMatch.getMasterSchedule();
+			if (masterSchedule.length() != 0) {
+
+				// Load the routes from the RouteMatch object
+				this.routes = Route.generateRoutes(masterSchedule);
+
+				// If there are routes that were loaded, execute the following:
+				if (this.routes.length != 0) {
+
+					// If polylines are enabled, execute the following:
+					if (SettingsPopupWindow.SHOW_POLYLINES) {
+
+						// Update the progress to 1/3, and inform the user that we are now loading polylines
+						Log.d("loadData", "Loading polylines");
+						this.setProgress(1d / 3d);
+						this.setMessage(R.string.load_polylines);
+
+						// Load the polyline coordinates into each route
+						for (int polylineIndex = 0; polylineIndex < this.routes.length; polylineIndex++) {
+							Route route = this.routes[polylineIndex];
+
+							// Load the polylineCoordinates into the route
+							route.polyLineCoordinates = route.loadPolyLineCoordinates(this.routeMatch);
+
+							// Set the progress to the current index (plus 1) out of the all the routes to be parsed,
+							// and then divide that value in third, as this is the other 33% to be processed.
+							this.setProgress((1d / 3d) + (((polylineIndex + 1d) / this.routes.length) / 3d));
+						}
+
+						// Update the progress to 2/3, and inform the user that we are now loading stops.
+						Log.d("loadData", "Loading stops in the routes");
+						this.setProgress(2d / 3d);
+						this.setMessage(R.string.load_stops);
+
+						// Load the stops in each route.
+						for (int i = 0; i < this.routes.length; i++) {
+							// Get the route at the current index (i) from all the routes that were loaded.
+							Route route = this.routes[i];
+
+							// Load the stops in the route.
+							route.stops = route.loadStops(this.routeMatch);
+
+							// Set the progress to the current index (plus 1) out of the all the routes to be parsed,
+							// and then divide that value in third, as this is the other 33% to be processed.
+							this.setProgress((2d / 3d) + (((i + 2d) / this.routes.length) / 3d));
+						}
+					} else {
```

Comment:
These sections are fairly similar. Consider moving these to a function, and somehow manage the progress?